asset-transfer-secured-agreement/application-gateway-typescript/src/app.ts

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { connect, Contract} from '@hyperledger/fabric-gateway';
+import { TextDecoder } from 'util';
+import crpto from 'crypto';
+import { newGrpcConnection,newIdentity,newSigner,tlsCertPathOrg1,peerEndpointOrg1,peerNameOrg1,certPathOrg1,mspIdOrg1,keyDirectoryPathOrg1,tlsCertPathOrg2,peerEndpointOrg2,peerNameOrg2,certPathOrg2,mspIdOrg2,keyDirectoryPathOrg2} from './connect'
+
+const utf8Decoder = new TextDecoder();
+
+const RED = '\x1b[31m\n';
+const GREEN = '\x1b[32m\n';
+const RESET = '\x1b[0m';
+
+const channelName = 'mychannel';
+const chaincodeName = 'secured';
+
+//Use a random key so that we can run multiple times
+const now = Date.now();
+const assetKey= `asset${now}`;
+
+//Generate random bytes using crypto
+const randomBytes = crpto.randomBytes(256).toString('hex')
```

Comment:
Using the same salt for every transaction that requires salt is not really best-practice. You could just make this a function and call it (as `randomBytes()`) for a new salt every time you need one.
```suggestion
function randomBytes(): string {
    return crpto.randomBytes(256).toString('hex');
}
```

Reply:
This actually failed the verify asset properties function as it is comparing the hash stored and the data passed.